src/root.zig

```diff
@@ -673,36 +705,40 @@ pub const Unicode = struct {
 
     alg: Algorithm,
     opts: Options,
-    gcd: GenCatData,
-    cd: CaseData,
+    unicode_toolbox: UnicodeToolBox,
```

Comment:
Okay I think I understand what you are doing here; you're reusing the same `NormData` instance for each call to `convertString`. That's fine (I think), but I don't think we need the `UnicodeToolBox` struct to do that, and instead just add another field to `Unicode` with the `norm_data` instance.

Reply:
The function `CharacterType.fromUnicode` need also `gcd` and `cd`. In my head, it was simpler to have a toolbox with all we can need to play with unicode.

Reply:
It is `pub` because the file `utils.zig` need to import it.

And yes, it is like a Russian doll, but I found it practical.
We can also add the four field directly to the `Unicode` if you prefer. You just have to say it.